lib/compress.js

```diff
@@ -1117,10 +1134,11 @@ merge(Compressor.prototype, {
             throw def;
         });
         def(AST_Dot, function(compressor){
-            if (compressor.option("unsafe") && this.property == "length") {
-                var str = ev(this.expression, compressor);
-                if (typeof str == "string")
-                    return str.length;
+            if (compressor.option("unsafe")) {
+                var val = ev(this.expression, compressor);
+                if (HOP(val, this.property)) {
```

Comment:
@kzc done in https://github.com/mishoo/UglifyJS2/pull/1425/commits/b363786d8b5b7d815197e56b68eebd8058cfcc9b - thanks!